lambda_strip_docx/lambda_function.py

```diff
@@ -0,0 +1,116 @@
+    "w": "http://schemas.openxmlformats.org/wordprocessingml/2006/main",
+}
+
+def _strip_forbidden_attributes(root: lxml.etree._Element) -> None:
+    """Remove forbidden attributes from XML elements."""
+    forbidden_attributes = ["w15:author", "w15:userId", "w:author", "w:initials"]
+    for attribute in forbidden_attributes:
+        attribute_namespace, _, attribute_name = attribute.partition(":")
+        for node in root.xpath(f"//*[@{attribute}]", namespaces=NAMESPACES):
+            node.attrib[f"{{{NAMESPACES[attribute_namespace]}}}{attribute_name}"] = REDACTION_STRING
+
+
+def _strip_forbidden_tags(root: lxml.etree._Element) -> None:
+    """Remove content from forbidden tags."""
+    forbidden_tags = ["cp:lastModifiedBy", "dc:creator"]
+    for tag in forbidden_tags:
+        for node in root.xpath(f"//{tag}", namespaces=NAMESPACES):
+            if node.text is not None:
+                node.text = REDACTION_STRING
+
+
+def _strip_docx_author_metadata_from_xml(xml_content: bytes) -> bytes:
+    """Process XML content to remove author metadata."""
+    try:
+        root = lxml.etree.fromstring(xml_content)
+        _strip_forbidden_attributes(root)
+        _strip_forbidden_tags(root)
+        return lxml.etree.tostring(root)
+    except lxml.etree.XMLSyntaxError:
+        # Return original content if XML parsing fails
+        return xml_content
+
+
+def strip_docx_author_metadata_from_docx(input_bytes: bytes) -> bytes:
+    """Strip author metadata from a DOCX file in bytes form."""
+    input_buffer = io.BytesIO(input_bytes)
+    output_buffer = io.BytesIO()
+
+    with ZipFile(input_buffer, "r") as archive_input, ZipFile(output_buffer, "w", compression=ZIP_DEFLATED, compresslevel=6) as archive_output:
+        for archive_filename in archive_input.namelist():
+            with archive_input.open(archive_filename, "r") as f:
+                xml_content = f.read()
+                processed_content = _strip_docx_author_metadata_from_xml(xml_content)
+                archive_output.writestr(archive_filename, processed_content)
+
+    return output_buffer.getvalue()
+
+def lambda_handler(event, context):
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+
+    try:
+        # Initialize S3 client
+        s3 = boto3.client('s3')
+
+        # Process each record in the S3 event
+        for record in event.get('Records', []):
+            try:
+                # Extract bucket name and object key from the S3 event
+                bucket_name = record['s3']['bucket']['name']
+                object_key = unquote_plus(record['s3']['object']['key'])
+
+                logger.info(f"Processing file: {object_key} from bucket: {bucket_name}")
+
+                # Check if it's a DOCX file
+                if not object_key.lower().endswith('.docx'):
+                    logger.warning(f"Skipping non-DOCX file: {object_key}")
+                    continue
+
```

Comment:
We should check that the docx does not have the document-processing tag of the current version